packages/grid/x-data-grid-pro/src/hooks/features/rowGrouping/useGridRowGrouping.tsx

```diff
@@ -280,11 +80,11 @@ export const useGridRowGrouping = (
       const currentModel = gridRowGroupingModelSelector(apiRef);
       if (currentModel !== model) {
         apiRef.current.setState(mergeStateWithRowGroupingModel(model));
-        updateRowGrouping();
+        setStrategyAvailability(apiRef, props.disableRowGrouping);
```

Comment:
Could we put this in an effect to avoid having to call `setStrategyAvailability` in all situations where the model may change? In practice what I want to say is to memoize the callback that determines if a strategy is available, then the effect would have this callback as dependency. When the reference changes, `unstable_setStrategyAvailability` is called.

Reply:
The problem is that `setStrategyAvailability` must be updated when the columns are updated, but it does not have the column state as an argument, because the first time we are calling it, the state is not ready yet.

I don't see how we could both initialize it in `useGridRowGroupingPreProcessors` and put it in an `useEffect` in `useGridRowGrouping`.

Reply:
I might be missing something but the effect will only be called when the state is already initialized and the pre-processors are registered. `gridRowGroupingSanitizedModelSelector` is used to get the number of items, so the length could a dependency.

Reply:
On the effect of `useGridRowGrouping` that calls `setRowGroupingModel` yes.
But for the `useFirstRender` of `useRowGroupingPreProcessing` that inits the strategy before the initialization of the rows state, the columns state is not ready so we can't pass `gridRowGroupingSanitizedModelSelector` as a dep.

We could separate the two behaviors but I'm not sure it's better.